pipit/readers/otf2_reader.py

```diff
@@ -190,70 +192,130 @@ def events_reader(self, rank_size):
             # columns of the DataFrame
             timestamps, event_types, event_attributes, names = [], [], [], []
 
-            # Note:
-            # 1. The below lists are for storing logical ids.
-            # 2. Support for GPU events has to be added and unified across readers.
+            # note: the below lists are for storing logical ids
             process_ids, thread_ids = [], []
 
+            # list of metric members of in the definitions
+            metric_names = set(
+                self.definitions.loc[
+                    self.definitions["Definition Type"] == "MetricMember"
+                ]["Attributes"]
+                .map(lambda attr: attr["name"])
+                .to_list()
+            )
+
+            # maps each metric to a list of its values
+            metrics_dict = {metric_name: [] for metric_name in metric_names}
+
+            # used to keep track of information about
+            # the most recent metrics that were read
+            prev_metric_time = -1
```

Comment:
each metric event happens before a corresponding enter or leave with the same timestamp

Reply:
https://scorepci.pages.jsc.fz-juelich.de/otf2-pipelines/doc.r4707/python/basics.html#metrics

documentation that might be helpful to understand otf2 metric events